bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+class FunctionSignature():
+    """
+    Class to determine a Function's Signature.
+    """
+    function_cache = {} # Variable to store previous functions so as to not rejit.
+    def __init__(self, rdf = None) -> None:
+        if rdf != None:
```

Comment:
Under which circumstances can `rdf` be none? We will always come from a method call on some RDataFrame object, and if we reach this point it's because that object is not None.

Reply:
Ok. SO the None was something I added, just in case I wanted to use the functions from the class on a function for purposes outside the purpose of using a function on RDF. 

The only reason RDF Is there is to determine if the function has dependencies on columns of the RDF.

Reply:
Then it's better if we make this explicit by changing the `__init__` signature

```suggestion
    def __init__(self, col_names: List[str]):
        self.col_names = col_names
```